server/storage/mvcc/kvstore_txn.go

```diff
@@ -151,6 +160,18 @@ func (tr *storeTxnRead) rangeKeys(ctx context.Context, key, end []byte, curRev i
 		limit = len(revpairs)
 	}
 
+	if tr.readMode == ConcurrentReadTxNoCopyMode && tr.tx == nil { // ConcurrentReadTxNoCopyMode
+		var tx backend.ReadTx
+		if limit >= ExpensiveReadLimit { // expensive request
+			tx = tr.s.b.ConcurrentReadTx()
+		} else {
+			tx = tr.s.b.ConcurrentReadTxNoCopy()
```

Comment:
I am not very clear why we create a new ReadTx at this location.

Reply:
Not a new one. We do not create  readtx at the beginning. We should create one when we know limit.